provider/datagen/src/transform/cldr/locale_canonicalizer/likely_subtags.rs

```diff
@@ -86,13 +87,13 @@ impl DataProvider<LikelySubtagsForScriptRegionV1Marker> for crate::DatagenProvid
     }
 }
 
-impl IterableDataProvider<LikelySubtagsForScriptRegionV1Marker> for crate::DatagenProvider {
+impl IterableDataProvider<LikelySubtagsForScriptRegionV1Marker> for DatagenProvider {
     fn supported_locales(&self) -> Result<Vec<DataLocale>, DataError> {
         Ok(vec![Default::default()])
     }
 }
 
-pub(crate) struct LikelySubtagsResources<'a> {
+pub struct LikelySubtagsResources<'a> {
```

Comment:
I don't want to use `pub(crate)` visibility anywhere in `crate::provider`, so that there is no accidental leakage and it's easy to move this into a separate crate. The top level `mod transform` is now private, so everything below it can be `pub`.

Reply:
I'd prefer that types remain `pub(crate)` unless they are actually being exported. You shouldn't need to walk up the module graph to check if something is actually public, and furthermore, `pub`s can be re-exported. `pub(crate)` is much less error-prone. I think we've discussed this before in ICU4X-TC.

> so that there is no accidental leakage

Given that `pub(crate)` is stricter than `pub`, I don't see how changing to `pub` would prevent "accidental leakage". Quite the opposite.

Reply:
I had `pub(in crate::provider)` everywhere but that's just pretty verbose, so I simplified to `pub` as the whole module is private.